core/domain/stats_domain.py

```diff
@@ -1675,16 +1627,16 @@ def get_learner_answer_info_id(cls, state_reference, entity_type):
 
         Args:
             state_reference: str. This is used to refer to a state
-                in an exploration or question. Like for an exploration the
+                in an exploration or question. For an exploration the
                 value will be equal to 'exp_id.state_name' & for question
                 this will be equal to 'question_id' only.
-            entity_type: str. The type of entity i.e 'exploration' or
-                'question '.
+            entity_type: str. The type of entity i.e ENTITY_TYPE_EXPLORATION or
+                ENTITY_TYPE_EXPLORATION which are declared in feconf.py.
 
         Return:
             learner_answer_info_id: str. The id generated by the function.
         """
-        learner_answer_info_id = (state_reference + '.' + entity_type + '.' +
+        learner_answer_info_id = (entity_type + '.' + state_reference + '.' +
```

Comment:
I think in an earlier comment I mentioned that we prefer interpolation.

Also you still need to take out the first two parts.

Reply:
Done